src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -1537,6 +1539,10 @@ castExpression
         )
     ;
 
+ switchExpression
+    :   "switch"^ LPAREN expression RPAREN switchBlock
```

Comment:
it should be 
```
"switch"^ LPAREN expression RPAREN LCURLY
                ( casesGroup )*
            RCURLY
```

as this commit do no functional changes

Reply:
From [JLS](https://docs.oracle.com/javase/specs/jls/se13/preview/switch-expressions.html#jep354-15.28): 
<ins>**SwitchExpression:**</ins>
_`switch` ( Expression ) SwitchBlock_

Isn't it a good idea to follow JLS when possible?

Reply:
it is how it should looks in third commit, but in second commit it should be as it was before, commit is "minor: ...."

Reply:
Oh, yes. I understand. Thanks for explaining that.

Reply:
Fixed.

Reply:
now in second commit, we have duplicates of 
```
"switch"^ LPAREN expression RPAREN LCURLY
                ( casesGroup )*
            RCURLY
```

please explain why you moved it up ?

Reply:
I moved it above the `expression` rule to avoid some ambiguity.

Reply:
if such move up is required to new switch feature - it should stay in functional fix. 
But in second commit you can move keep moving in separate rule.

Reply:
Fixed.